tests/nack.rs

```diff
@@ -10,6 +10,163 @@ use str0m::{Event, RtcError};
 mod common;
 use common::{connect_l_r, init_log, progress};
 
+use crate::common::progress_with_loss;
+
+#[test]
+pub fn loss_recovery() -> Result<(), RtcError> {
+    init_log();
+
+    let (mut l, mut r) = connect_l_r();
+
+    let mid = "vid".into();
+
+    // In this example we are using MID only (no RID) to identify the incoming media.
+    let ssrc_tx: Ssrc = 42.into();
+    let ssrc_rtx: Ssrc = 44.into();
+
+    l.direct_api().declare_media(mid, MediaKind::Video);
+
+    l.direct_api()
+        .declare_stream_tx(ssrc_tx, Some(ssrc_rtx), mid, None);
+
+    r.direct_api().declare_media(mid, MediaKind::Video);
+
+    r.direct_api()
+        .expect_stream_rx(ssrc_tx, Some(ssrc_rtx), mid, None);
+
+    let max = l.last.max(r.last);
+    l.last = max;
+    r.last = max;
+
+    let params = l.params_vp8();
+    let ssrc = l.direct_api().stream_tx_by_mid(mid, None).unwrap().ssrc();
+    assert_eq!(params.spec().codec, Codec::Vp8);
+    let pt = params.pt();
+
+    let exts = ExtensionValues::default();
+
+    let to_write = &[0x1, 0x2, 0x3, 0x4];
+    let num_packets: usize = 1000;
+
+    // write all packets num_packets
+    for index in 0..num_packets {
+        let wallclock = l.start + l.duration();
+
+        let mut direct = l.direct_api();
+        let stream = direct.stream_tx(&ssrc).unwrap();
+
+        let time = (index * 1000 + 47_000_000) as u32;
+        let seq_no = (47_000 + index as u64).into();
+
+        stream
+            .write_rtp(
+                pt,
+                seq_no,
+                time,
+                wallclock,
+                false,
+                exts,
+                true,
+                to_write.to_vec(),
+            )
+            .expect("clean write");
+
+        if !(10..=990).contains(&index) {
```

Comment:
In the `src/lib.rs` I turned off this utterly insane lint. I think what you had before was more readable. Though I don't care hugely in tests.

Reply:
agree